pkg/pillar/zedcloud/authen.go

```diff
@@ -214,6 +157,10 @@ func VerifyAuthContainer(ctx *ZedCloudContext, sm *zauth.AuthContainer) (types.S
 	}
 
 	data := sm.ProtectedPayload.GetPayload()
+	if data == nil {
+		return types.SenderStatusNone, nil
+	}
```

Comment:
In what case do you have an empty payload?

Reply:
For the encrypted config the payload is nullified (the payload is in the cipher block). 
But the `VerifyAuthContainer()` is called for the encrypted case before actual decryption to verify fields of the AuthContainer: server cert is not missing, sender cert hash is correct, algorithms are correct.

So basically this says the following: verification won't expect (and won't crash) that payload is defined, but header fields should be defined and correct.